build/cargo.gni

```diff
@@ -29,9 +31,41 @@ template("cargo_build") {
       args += [ "--mac_deployment_target=" + mac_deployment_target ]
     }
 
-    if (defined(invoker.rust_flags)) {
+    if (defined(ldflags)) {
+      rust_flags += " -C link-args=${ldflags} "
+    }
+
+    if (defined(rust_flags)) {
+      print("rust flags", rust_flags)
       args += [ "--rust_flags=$rust_flags" ]
     }
 
   }
 }
+
+# Runs a script to generate C bindings for rust library
+template("cargo_cbindgen") {
+  action(target_name) {
+    script = "//brave/script/cargo_cbindgen.py"
+
+    forward_variables_from(invoker, ["config", "sources", "output"])
+
+    assert(defined(config), "Must specify cbindgen.toml config to use")
+    assert(defined(sources), "Must specify sources to generate bindings for")
+    assert(defined(output), "Must specify output header file")
+
+    sources += [
+      config,
+    ]
+    outputs = [ "$target_gen_dir/$output" ]
+
+    args = [
+      "--rustup_path=" + rebase_path("//brave/build/rustup/", root_build_dir),
```

Comment:
we should probably make rustup_path/cargo_path/etc.. vars in this gni file so we don't duplicate `rebase_path("//brave/build/rustup/", root_build_dir)` in other places